src/lib/components/Flowbite/ThemeContext.tsx

```diff
@@ -0,0 +1,70 @@
+/* eslint-disable react-hooks/rules-of-hooks */
+import { FC, ReactNode, createContext, useContext, useState, useEffect } from 'react';
+import defaultTheme from '../../theme/default';
+
+export type Mode = string | undefined | 'light' | 'dark';
+
+interface ThemeContextProps {
+  theme: any;
+  mode?: Mode;
+  toggleMode?: () => void | null;
+}
+
+export const ThemeContext = createContext<ThemeContextProps>({
+  theme: defaultTheme,
+});
+
+interface ThemeProviderProps {
+  children: ReactNode;
+  value?: any;
+}
+
+export const ThemeProvider: FC<ThemeProviderProps> = ({ children, value }) => {
+  return <ThemeContext.Provider value={value}>{children}</ThemeContext.Provider>;
+};
+
+export function useTheme(): ThemeContextProps {
+  return useContext(ThemeContext);
+}
+
+export const useThemeMode = (
+  usePreferences: boolean,
+): [Mode, React.Dispatch<React.SetStateAction<Mode>> | undefined, (() => void) | undefined] => {
+  if (!usePreferences) return [undefined, undefined, undefined];
+  const [mode, setMode] = useState<Mode>(undefined);
+
+  const savePreference = (m: string) => window.localStorage.setItem('theme', m);
```

Comment:
In theory you should need to check for `window` for all of these in SSR apps but I haven't noticed any issues in Remix

Reply:
well, I can access the localStorage directly, without use window...